apps/managedidentity/managedidentity_test.go

```diff
@@ -656,6 +659,112 @@ func TestAppServiceAcquireTokenReturnsTokenSuccess(t *testing.T) {
 		})
 	}
 }
+
+func TestAzureMLAcquireTokenReturnsTokenSuccess(t *testing.T) {
+	setEnvVars(t, AzureML)
+	testCases := []struct {
+		resource string
+		miType   ID
+	}{
+		{resource: resource, miType: SystemAssigned()},
+		{resource: resourceDefaultSuffix, miType: SystemAssigned()},
+		{resource: resource, miType: UserAssignedClientID("clientId")},
+	}
+	for _, testCase := range testCases {
+		t.Run(string(AzureML)+"-"+testCase.miType.value(), func(t *testing.T) {
+			endpoint := "http://127.0.0.1:41564/msi/token"
+
+			var localUrl *url.URL
+			mockClient := mock.Client{}
+			responseBody, err := getSuccessfulResponse(resource, false)
+			if err != nil {
+				t.Fatalf(errorFormingJsonResponse, err.Error())
+			}
+			mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK),
+				mock.WithBody(responseBody),
+				mock.WithCallback(func(r *http.Request) {
+					localUrl = r.URL
+				}))
+			// resetting cache
+			before := cacheManager
+			defer func() { cacheManager = before }()
+			cacheManager = storage.New(nil)
+
+			client, err := New(testCase.miType, WithHTTPClient(&mockClient))
+			if err != nil {
+				t.Fatal(err)
+			}
+			result, err := client.AcquireToken(context.Background(), testCase.resource)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if localUrl == nil || !strings.HasPrefix(localUrl.String(), endpoint) {
+				t.Fatalf("url request is not on %s got %s", endpoint, localUrl)
+			}
+			query := localUrl.Query()
+
+			if query.Get(apiVersionQueryParameterName) != azureMLAPIVersion {
+				t.Fatalf("api-version not on %s got %s", azureMLAPIVersion, query.Get(apiVersionQueryParameterName))
+			}
+			if r := query.Get(resourceQueryParameterName); strings.HasSuffix(r, "/.default") {
+				t.Fatal("suffix /.default was not removed.")
+			}
+			if result.Metadata.TokenSource != base.IdentityProvider {
+				t.Fatalf("expected IdentityProvider tokensource, got %d", result.Metadata.TokenSource)
+			}
+			if result.AccessToken != token {
+				t.Fatalf("wanted %q, got %q", token, result.AccessToken)
+			}
+			if testCase.miType.value() == "clientId" {
+				if query.Get("clientid") != "clientId" {
+					t.Fatalf("expected clientid to be set to clientId, got %s", query.Get("clientid"))
+				}
+			} else {
+				if query.Get("clientid") != os.Getenv("DEFAULT_IDENTITY_CLIENT_ID") {
+					t.Fatalf("expected clientid to be set to default identity client id, got %s", query.Get("clientid"))
+				}
+			}
```

Comment:
There are 2 cases to test here:
- DEFAULT_IDENTITY_CLIENT_ID = A & SystemAssigned() -> clientid=A
- DEFAULT_IDENTITY_CLIENT_ID = A & UserAssignedClientID("B") -> clientid=B

This doesn't cover them because you don't set the variable, and both `Get` and `Getenv` return `""` when the given key has no value